File: Solved/Dang O Dong/Dang o Dong.py
# دَنگ و دُنگ
# ID : 127290
# https://quera.org/problemset/127290/
# https://b2n.ir/j65197

a = int(input())
h = []

for i in range(a):
    b = input().split()
    q = int(b[0])
    q1 = int(b[1])
    q2 = int(b[2])
    q1 -= q2

    if q1 % q == 0 and (q1 / q != 0) and (q1 / q > 0):
        res = q1 // q

    elif q1 / q == 0:
        res = (-1)

    elif q1 % q != 0:
        res = -1

    elif q1 % q == 0 and (q1 / q < 0):
        res = -1

    h.append(res)

for j in range(len(h)):
    k = h[j]
    print(k)

File: Solved/Dang O Dong/DangODong.java
import java.util.Scanner;

/*
Question title = Dang va Dong - دَنگ و دُنگ
https://quera.ir/problemset/127290/
 */

public class DangODong {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        int rows = Integer.parseInt(sc.nextLine());
        StringBuilder result = new StringBuilder();
        for (int i = 1; i <= rows; i++) {

            String[] line = sc.nextLine().split("\\s");
            result.append(result(line)).append("\n");

        }
        System.out.println(result.toString());
    }

    private static String result(String[] line) {
        int people = Integer.parseInt(line[0]);
        int paidMoney = Integer.parseInt(line[1]);
        int fee = Integer.parseInt(line[2]);

        int paidPerPerson = (paidMoney - fee) / people;
        if (paidPerPerson <= 0) {
            return "-1";
        }
        int paidByPayer = paidMoney - ((people - 1) * paidPerPerson);
        int paidPerPersonWithFee = paidPerPerson + fee;
        if (paidByPayer == paidPerPersonWithFee)
            return paidPerPerson + "";
        else
            return "-1";
    }
}